### ISSUE TITLE
GM_xmlhttpRequest Cookie not sent

### ISSUE THREAD

**Issue by friedrico** [greasemonkey/greasemonkey#2181]
Hi
I use the following snippet (Greasemonkey 3.1 (23.3.2015))
```javascript
GM_xmlhttpRequest({
        method: "GET",
        url: " https://www.example.com/members-area",
        onload: function(response) {
                console.log(response.responseText);
        }
    });
}
```
The problem is that the responseText is not what I'd expect since it contains a login form. The reason is probably that the cookie isn't sent. I guess it is a httponly cookie - so could that be the reason? Would there be a way of setting/sending that cookie automatically?

Setting it manually works perfectly (wireshark + headers:{'Cookie':...})

No possible value for network.cookie.cookieBehavior helps either.

The goal is just to make requests within my member's area

**Comment by janekptacijarabaci** [comment]
See:
https://github.com/greasemonkey/greasemonkey/issues/1521#issuecomment-4419103
https://github.com/greasemonkey/greasemonkey/issues/1521#issuecomment-4454812

> Every request is cross origin because they all start in the privileged chrome origin (resp. the XPCOM component), which can access any remote url -- but then they are all "third party".

**Comment by friedrico** [comment]
Yeah I was aware of that post. It explains why, but doesn't solve the problem

**Comment by janekptacijarabaci** [comment]
Yes. I think the solution would be this:
https://github.com/greasemonkey/greasemonkey/compare/master...janekptacijarabaci:GM_xmlhttpRequest_cookies_3rdParty

**Comment by janekptacijarabaci** [comment]
Can I do pull request?

**Comment by arantius** [comment]
I already plan on merging the branch.

**Comment by arantius** [comment]
Fix should be in 3.2beta3:

https://addons.mozilla.org/firefox/downloads/file/318054/greasemonkey-3.2beta3-fx.xpi?src=devhub

Confirmation would be nice!

**Comment by friedrico** [comment]
Just installed and it worked perfectly.
Thank you guys
